src/lms-ds-loader/edfi_lms_ds_loader/migrator.py

```diff
@@ -53,57 +34,45 @@ def _read_statements_from_file(full_path: str) -> List[str]:
     with open(full_path) as f:
         raw_sql = f.read()
 
-    return split(raw_sql)
+    statements: List[str] = split(raw_sql)
+    return statements
 
 
-def _execute_statements(engine: Engine, statements: List[str]):
-    def __callback(session: sa_session):
-        for statement in statements:
-            # Ignore MSSQL "GO" statements
-            if statement == "GO":
-                continue
-
-            # Deliberately throwing away all results. Counting on exception handling
-            # if there are any errors, and migration scripts should not be returning
-            # any results.
-            session.execute(statement)  # type: ignore
-
-    _execute_transaction(engine, __callback)
-
-
-def _script_has_been_run(engine: Engine, migration: str) -> bool:
-    def __callback(session: sa_session):
-        statement = f"SELECT 1 FROM lms.migrationjournal WHERE script = '{migration}';"
-        return session.execute(statement).scalar()  # type: ignore
-
+def _script_has_been_run(engine: sa_Engine, migration: str) -> bool:
     try:
-        response = _execute_transaction(engine, __callback)
+        statement = f"SELECT 1 FROM lms.migrationjournal WHERE script = '{migration}';"
+        response = get_int(engine, statement)
```

Comment:
I tried monkey patching `get_int` but it did not work. Now that I'm reviewing the code, I wonder if it is something about the import statement running before the mock is generated - thus the mock is ignored. I seem to recall something about that when I was working more closely on this project last November or December.

Tried `mocker.patch("edfi_lms_ds_loader.sql_adapter.get_int", 0` and also tried injecting a method instead of a number, and it was always ignored.

Reply:
I think you're right! That sounds very familiar and I think I had learned that last fall, but forgotten about it by now.